### PR TITLE
DevKit updates for master branch

### PR DESCRIPTION
**Pull request by ElectricMaxxx**

### CODE REVIEW

**Comment by dbu** [comment]
lets discuss things in https://github.com/symfony-cmf/routing-bundle/pull/434#pullrequestreview-197406052 and then update this once the questions have been decided.

**Comment by ElectricMaxxx** [comment]
@dbu you know a minimum version of `Frameworkbundle` to avoid my issue:
```
PHP Fatal error:  Declaration of Symfony\Bundle\FrameworkBundle\Test\KernelTestCase::tearDown() must be compatible with PHPUnit\Framework\TestCase::tearDown(): void in /home/travis/build/symfony-cmf/Testing/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Test/KernelTestCase.php on line 136
```

File: src/Phpunit/DatabaseTestListener.php [hunk 1]
```diff
@@ -1,225 +0,0 @@
...
-        switch (static::$currentSuite->getName()) {
-            case 'orm':
-                $db = $test->getDbManager('ORM');
-                $purger = new Purger\ORMPurger($db->getOm());
-
-                break;
-
-            case 'phpcr':
-                $db = $test->getDbManager('PHPCR');
-                $purger = new Purger\PHPCRPurger($db->getOm());
```
Comment (dbu) [inline src/Phpunit/DatabaseTestListener.php]: should we not keep purging the database between tests? i agree that the setup is better done with the makefile, but things done between each test can not be moved to the makefile. we could provide a base test or trait of course, but not sure if that is better.

**Comment by dbu** [comment]
i'd take the newest 3.8 and 4.1 versions in composer.json then.

can we move to phpunit 7 while we are at it?

**Comment by ElectricMaxxx** [comment]
This goes green now :-) @dbu i would like to get your review, again pls.

File: src/Phpunit/DatabaseTestListener.php (hunk 1 above)
Comment (ElectricMaxxx) [reply src/Phpunit/DatabaseTestListener.php]: We could keep that possibibility, to use it, but none of our packages currently uses it.

Comment (ElectricMaxxx) [reply src/Phpunit/DatabaseTestListener.php]: I would have no problem to keep it, i just would have to make some minor tweaks to stay compatible with the interface

**Pull request merged**